src/main/java/htsjdk/samtools/HtsgetBAMFileReader.java

```
@@ -0,0 +1,561 @@
+     * @param samRecordFactory     SAM record factory
+     * @param useAsynchronousIO    if true, use asynchronous I/O
+     */
+    public HtsgetBAMFileReader(final URI source,
+                               final boolean eagerDecode,
+                               final ValidationStringency validationStringency,
+                               final SAMRecordFactory samRecordFactory,
+                               final boolean useAsynchronousIO) throws IOException {
+        this(source, eagerDecode, validationStringency, samRecordFactory, useAsynchronousIO, BlockGunzipper.getDefaultInflaterFactory());
+    }
+
+    /**
+     * Prepare to read BAM from a htsget source
+     *
+     * @param source               source of bytes.
+     * @param eagerDecode          if true, decode all BAM fields as reading rather than lazily.
+     * @param validationStringency Controls how to handle invalidate reads or header lines.
+     * @param samRecordFactory     SAM record factory
+     * @param useAsynchronousIO    if true, use asynchronous I/O
+     * @param inflaterFactory      InflaterFactory used by BlockCompressedInputStream
+     */
+    public HtsgetBAMFileReader(final URI source,
+                               final boolean eagerDecode,
+                               final ValidationStringency validationStringency,
+                               final SAMRecordFactory samRecordFactory,
+                               final boolean useAsynchronousIO,
+                               final InflaterFactory inflaterFactory) throws IOException {
+        this.mSource = source;
+        this.mEagerDecode = eagerDecode;
+        this.mValidationStringency = validationStringency;
+        this.mSamRecordFactory = samRecordFactory;
+        this.mUseAsynchronousIO = useAsynchronousIO;
+        this.mInflaterFactory = inflaterFactory;
+
+        final HtsgetRequest req = new HtsgetRequest(this.mSource).withDataClass(HtsgetClass.header);
```

We should always be setting the format here to BAM shouldn't we?  Otherwise we might get back cram header/ blocks which we can't parse? 
If we set the format to BAM is the server allowed to return CRAM anyway?

---

BAM is the default so it should be interpreted as BAM without setting it explicitly. The spec says the server SHOULD return an error if it can't serve the request format, so no